components/Navbar.js

```diff
@@ -1,109 +1,90 @@
-// import Image from 'next/image';
 import Link from 'next/link';
-import React from 'react';
-// import gmData from '../../config/gmData';
+import { useRouter } from 'next/router';
+import React, { useState, useEffect } from 'react';
 
-//to allow GM page to be visible on the navbar and accessible
-//uncomment the above import line "import gmData from '../../config/gmData';" to get quarter from the gmData page
-//uncomment the two lines saying  "/* <Link href="/gm"><li>{gmData.date.quarter} gm</li></Link> */"
-
-function MobileNavItem() {
-	return (
-		<li id="committee-mobile-nav-item">
-			<ul className="committee-mobile-nav">
-				<li><Link href={'/committees#studio'}><a>Studio</a></Link></li>
-				<li><Link href={'/committees#icpc'}><a>ICPC</a></Link></li>
-				<li><Link href={'/committees#design'}><a>Design</a></Link></li>
-				<li><Link href={'/committees#cyber'}><a>Cyber</a></Link></li>
-				<li><Link href={'/committees#teachla'}><a>Teach LA</a></Link></li>
-				<li><Link href={'/committees#w'}><a>W</a></Link></li>
-				<li><Link href={'/committees#ai'}><a>AI</a></Link></li>
-				<li><Link href={'/committees#hack'}><a>Hack</a></Link></li>
-			</ul>
-		</li>
-	);
-}
+export default function Navbar() {
+  // set states
+  const [menuActive, setMenuActive] = useState('');
+  const [linkActive, setLinkActive] = useState('');
 
+  const router = useRouter();
 
-export default class Navbar extends React.Component {
-	constructor(props) {
-		super(props);
-		this.clickMobileNav = this.clickMobileNav.bind(this);
-	}
+  // switches mobile menu state
+  const menuActivate = () => {
+    setMenuActive(menuActive ? '' : 'active');
+  };
 
-	clickMobileNav() {
-		// TODO: remove direct DOM manipulation, make this stateful
-		document.getElementById('menu-toggle').checked = false;
-	}
+  // allows accessible access to mobile menu
+  const menuAccessible = (event) => {
+    if (event.code === 'Enter' || event.code === 'Space') {
+      setMenuActive(menuActive ? '' : 'active');
+    }
+  };
 
-	// TODO: can we both refactor this to use useEffect, and/or remove
-	// this listener entirely
+  useEffect(() => {
+    const hamburger = document.querySelector('#hamburger');
+    const navItems = document.querySelector('.nav-items');
+    const listItems = navItems.querySelectorAll(':scope > li');
 
-	componentDidMount() {
-		// TODO: remove direct DOM manipulation, make this stateful
-		const items = document.querySelectorAll('#mobile-nav .nav-items li');
-		for (const item of items) {
-			item.addEventListener('click', this.clickMobileNav);
-		}
-	}
+    setLinkActive(router.asPath);
+    hamburger.addEventListener('click', menuActivate);
```

Comment:
Instead of adding this through the DOM (which I'd like to avoid if possible), could we instead bind this to the corresponding JSX element?

i.e.

```jsx
<div className={menuActive} id="hamburger" 
// ...
onClick={menuActivate} // or whatever you end up naming this as
>
```

That way, we can avoid manipulating the DOM outside of React, _and_ we don't need to write our own cleanup code!